SpatialGDK/Source/SpatialGDK/Private/Interop/SnapshotManager.cpp

```diff
@@ -13,18 +12,24 @@ DEFINE_LOG_CATEGORY(LogSnapshotManager);
 
 using namespace SpatialGDK;
 
-void USnapshotManager::Init(USpatialNetDriver* InNetDriver)
+SpatialSnapshotManager::SpatialSnapshotManager()
+	: Connection(nullptr)
+	, GlobalStateManager(nullptr)
+	, Receiver(nullptr)
+{}
+
+void SpatialSnapshotManager::Init(USpatialWorkerConnection* InConnection, UGlobalStateManager* InGlobalStateManager, USpatialReceiver* InReceiver)
 {
-	NetDriver = InNetDriver;
-	Receiver = InNetDriver->Receiver;
-	GlobalStateManager = InNetDriver->GlobalStateManager;
+	Connection = InConnection;
+	Receiver = InReceiver;
+	GlobalStateManager = InGlobalStateManager;
 }
 
 // WorldWipe will send out an expensive entity query for every entity in the deployment.
 // It does this by sending an entity query for all entities with the Unreal Metadata Component
 // Once it has the response to this query, it will send deletion requests for all found entities.
 // Should only be triggered by the worker which is authoritative over the GSM.
-void USnapshotManager::WorldWipe(const USpatialNetDriver::PostWorldWipeDelegate& PostWorldWipeDelegate)
+void SpatialSnapshotManager::WorldWipe(const PostWorldWipeDelegate& PostWorldWipeDelegate)
```

Comment:
Follow up on PostWorldWipeDelegate. I assume SnapshotManager.cpp knows what PostWorldWipeDelegate is because one of the headers this file includes also includes SpatialNetDriver? Perhaps it would be good to move the delegate decoration into SnapshotManager.h? It seems as if SnapshotManager still relies on having the SpatialNetDriver header.

Reply:
Ah, you're right. I'll think about the proper place for `PostWorldWipeDelegate`

Reply:
Added to summary below.